README
XTM::*
---------

This package contains Perl extension interfaces for Topic Maps

   http://www.topicmaps.org/xtm/1.0/

To build the extensions, unpack this distribution somewhere, and
exercise the usual incantation:

perl Makefile.PL
make
make test
make install

You will need perl version 5.005 or better to install these modules.
Further documentation is embedded in the individual modules.

See

   http://topicmaps.bond.edu.au/

for up-to-date information about the package.

Copyright 2001 Robert Barta

This library is free software; you can redistribute it and/or
modify it under the same terms as Perl itself.


Changes
Revision history for Perl extension XTM.

0.37 rho Sun Jun  5 19:23:20 EST 2005
	- added XTM::AsTMa::Converter to do AsTMa= -> XTM conversion

0.36 rho Wed Mar  9 19:29:21 EST 2005
	- added patch from Niko Tyni <ntyni@iki.fi> to make it work under Perl 5.8.4 (MUCHAS GRACIAS)
	- removed situations which created warnings under Perl 5.8
     az Sun Feb 29 16:22:34 2004
	- added here-docs to astma: 
	'in: \token' includes all stuff up to '\token'
	- added test for here-docs to 04astma
	- removed inline comments from astma

0.35  Mon Aug  5 21:20:43 EST 2002
	- added missing maps/test.ltm (need better cpan simulator)

0.34  Sun Aug  4 21:32:45 EST 2002
	- added LTM/CParser (would not build automatically on some systems?)
	- more speed improvements in XTM::Memory (topics query)
	- fixed documentation bug for induced_topic_tree
	- added class-instance as special association type to induced_topic_tree
	- added loop support (via shared trees) in induced_topic_tree, pruning reduces runtime
	- added some PSIs to XTM::PSI
	- added method descriptions to the most important base classes

0.33  Fri Jul 26 21:35:17 EST 2002
	- fixed bug which raises exception when XTM::Path creating /topic[@id = '...']
	- changed simple search syntax in XTM::Memory 'regexp' -> 'regexps'
	- added commands 'merge', 'duplicate_suppression', 'follow_maps' to xtm.pl
	- added consistency to be passed through all sync_ins also through XTM::Virtual
	- added t/08topic.t for simple tests
	- added 'is-reified-by' clauses in AsTMa (XTM::AsTMa::Parser) for topics and associations
	- added AsTMa directive auto_complete
	- added AsTMa directive encoding to occur several times in a document
	- added first (still experimental) version of LTM parsing, see XTM::LTM
	- fixed strange 'cannot add to undefined' error in XTM::Path when using XTM::Path::create and find inside one program (workaround, not solved yet)
	- allowed several scope topics for an association (follows XTM and LTM)
	- added caching to XTM::Path, already parsed expressions are cached
	- added variable to XTM::Path (similar to SQL prepare/expecute)
	- added method add_defaults to XTM::topic to add default type/scope to topic components if missing
	- modified Memory, generic for major performance improvements when merging (still some potential to be used)
	- modified Memory: merging dies if a topic is merged with an existing but both have a
	     different addressable resource
	- fixed bug in XTM::Path not detecting correctly strings '0'

0.32  Sun Jun 30 20:36:01 EST 2002
	- added XTM::grammar, a package to hold element dependencies and nesting (internal mainly)
	- added XTM::Path, XPath-like navigation through XTM::base data structures
	- XTM::Memory::id: setting of id added
	- XTM::topic, doc corrections
	- using a precompiled version of the Parse::Recdescent (considerable AsTMa= speedup)
	- fixed potential problems in XTM::generic
	- added merging code to XTM::Memory, is controlled via a hash in XTM constructor (BACKWARD COMPATIBILITY?)
	- fixed docu/code mismatch in XTM::topic (occurrences)
	- added scope_eq to XTM::scope
	- added canonicalize to XTM::topic (removes duplicates, sorts things)
	- added more query clauses for the simple query language in XTM::Memory
	- more directives for XTM::AsTMa (naming, encoding). Use with care.

0.31  Thu May 16 16:56:40 EST 2002
	- got rid of XTM::XML::ParseError and used XML::SAX::Exception instead	

0.30  Wed May  8 17:50:02 EST 2002
	- removed dependency of File::Slurp (weird version number), reported by az
	- fixed bug in incorrectly reporting line numbers in AsTMa::Parser, reported by az
	- various documention bugs fixed (still much to do)
	- astma2xtm.pl: some minor changes in switches (see man page)
	- AsTMa parser: local comments MUST have at least one blank before the '#' (see docs, may break some maps, sorry)
	- activated tests (old XML::LibXML core dump error has disappeared)

0.29  Feb 25 07:23 EST 2002
	- Debian test version

0.28  Wed Jan 16 22:17:57 EST 2002
	- fixed empty character handling in AsTMa::Parser, added line number for better debugging
	- added more accurate dependencies
	- added some explanation to the README
	- fixed XTM::XML::MemoryBuilder to work with SAX::Expat, LibXML::SAX and PurePerl (still a problem there)
	- AsTMa parser: added directives %cancel, %log

0.27  Mon Jan 14 12:23:41 EST 2002
	- added missing XTM::AsTMA::MemoryBuilder
	- fixed test suite bugs which occur at some other Perl installations

0.26  Mon Dec 31 21:49:37 EST 2001
	- add xml output serialization for (most) classes
	- revamped XML::ATM into XML::AsTMa, XML::ATM is not distributed any more
	- added astma2xtm.pl into bin/ (converter), this obsoletes atm2xtm.pl
	- fixed some bugs in XTM::XML
	- added connected methods to topic and association to return all topic refs mentioned there
	- added test for topic IDs in XTM::topic.pm (experimental), have to be valid XML identifiers
	- started to move to Test::More for testing
	- added bin/prettifyxtm.pl
	- fixed unclean code in XTM::topic::has_instanceOf, checking for URIs now
	- moved XML parsing to XML::LibXML::SAX using the XML::SAX factory
	- moved most testing to Test::More, added numerous tests
	- added population mechanism into XTM::topic to load default values
	- added auto_complete parameter to all XTM::IO subclasses to control auto_completion, default is 1
	- updated README to reflect Perl version 5.6.1
	- added support rt.cpan.org to README 
	- added 'stats' command to xtm.pl
	- added commands 'topics' and 'assocs' to xtm.pl
	- modified URIs handling in xtm.pl: URIs without a scheme will default as file: URIs

0.25  Mon Dec 17 18:06:24 EST 2001
	- added statistics to info() in XTM
	- added clusters() in XTM

0.24  Sat Oct 20 18:46:32 EST 2001
	- fixed bug of not escaping & in XTM::ATM for occurrences
	- fixed bug of not checking for all URI schemes in XTM::ATM
	- fixed too restrictive pattern in XTM::ATM to allow valid XML ids as topic ids

0.23  Sat Oct 13 12:32:11 EST 2001

lib/XTM/base.pm
package XTM::base;

# $Id: base.pm,v 1.5 2005/06/05 09:22:43 rho Exp $ 

use strict;
use vars qw($VERSION);
$VERSION = '0.37';

=pod

=head1 NAME

XTM - Topic Map, base distribution

=head1 SYNOPSIS

  use XTM;
  $tm = new XTM; # creating an empty map

  # reading a topic map description from an XML file
  use XTM::XML;
  $tm = new XTM (tie => new XTM::XML (file => 'mymap.tm'));

  # application specific persistent medium
  $tm = new XTM (tie => new XTM::Application::MineMineMine);

=head1 DESCRIPTION

Topic Maps are a means for layering multidimensional topic spaces on top of
information assets. The standard covers concepts like topics,
associations, scopes and occurrences. Topic Maps are
expected to have a major impact on future information systems and compete
with RDF in the semantic web arena.

See http://www.topicmaps.org/xtm/1.0/ for more information.

Topic maps can be loaded from an XML based resource (local or remote
via an URL) as prescribed by the XTM standard. There is also a way
to tie a map to any persistent medium. The latter is especially interesting
when running this module along with a (web) server.

=head1 PROGRAMS

Following programs come with this distribution:

=over 4

=item L<xtm>.pl

A simple text interface to load & query topic maps locally.

=item L<astma2xtm>.pl

An AsTMa= to XTM converter.

=item L<prettifyxtm>.pl

A filter which formats any incoming XTM stream into a nested XML document.

=back

=head1 PACKAGES

Following packages come with this distribution:

=over 4

=item L<XTM>

toplevel wrapper for (non-threaded) maps

=item L<XTM::Memory>

in-memory representation of a map

=item L<XTM::PSI>

PSI definitions

=item L<XTM::IO>

Abstract class

=item L<XTM::AsTMa>

AsTMa adapter, this allows you to bind maps to an external AsTMa instance

=item L<XTM::LTM>

LTM adapter, this allows you to bind maps to an external LTM instance

=item L<XTM::XML>

XML adapter, this allows to to bind maps to an external XTM document

=item L<XTM::XML::MemoryBuilder>

SAX filter to create in-memory representation of a TM

=item L<XTM::Log>

logging facility will be used by others

=item L<XTM::generic>

accessor functions for all below

=item L<XTM::topic>

implementation of the 'topic' class

=item L<XTM::association>

implementation of the 'association' class

=item L<XTM::topicRef>

generic class definition

=item L<XTM::baseName>

generic class definition

=item L<XTM::scope>

generic class definition

=item L<XTM::resourceRef>

generic class definition

=item L<XTM::instanceOf>

generic class definition

=item L<XTM::subjectIdentity>

generic class definition

=item L<XTM::subjectIndicatorRef>

generic class definition

=item L<XTM::variant>

generic class definition

=item L<XTM::variantName>

generic class definition

=item L<XTM::parameters>

generic class definition

=item L<XTM::roleSpec>

generic class definition

=item L<XTM::member>

generic class definition

=item L<XTM::mergeMap>

generic class definition

=item L<XTM::resourceData>

generic class definition

=item L<XTM::baseNameString>

generic class definition

=item L<XTM::occurrence>

generic class definition

=back

=head1 AUTHOR INFORMATION

Copyright 200[1-25], Robert Barta <rho@telecoma.net>, All rights reserved.

This library is free software; you can redistribute it
and/or modify it under the same terms as Perl itself.
http://www.perl.com/perl/misc/Artistic.html

=cut


1;

__END__
